solver/examples/fistaExample.cpp

```diff
@@ -55,15 +55,14 @@ int main(int argc, char** argv) {
     /**
      * Set up the weights for the grid.
      */
-    auto weights = DataVector(grid->getSize());
-    weights.setAll(0.0);
+    DataVector weights(grid->getSize());
     double lambda = 1.0;
     const double gamma = 0.98;
     /**
      * Iterate over a few regularization penalties.
      */
     for (int i = 0; i < 5; ++i) {
-        weights.setAll(0.0);
```

Comment:
I understand that the example is meant to show how this could be extended to a more generic use-case, so that's why I left it as commented out, but setting the vector to 0 5 times on tp of the instantiation (which sets it to 0) is just extra useless work, isn't it? It changes nothing of the logic of the problem or the outcome

Reply:
My understanding was that the solver.solve call on line 76 takes weights as an initial guess _and_ writes the result back into it. it probably makes a difference in later iterations whether the initial guess is 0 or not.

Reply:
Sure, but a few lines above (i.e. line 58) `weights` is actually created as a `DataVector,` thus being initialized with zeros :grin: So the `setAll` in this for loop is just extra work... Or am I missing something?

Reply:
Here's what I mean:

Iteration:
1. weights is zero initialized, solve.solve gets called with zero weights as input but writes back result into weights as well!
2. weights is now not zero anymore, solve.solve gets called again with weights as input which are now different
3. etc.

In the old version solve.solve would've always been called with weights being zero. This should make difference no?

Reply:
Oh, I must be completely blind... 3 times I looked at this code and only now I've noticed the for loop is longer than I thought... yes, you are obviously right. The comments indentation somehow threw me off.

And that's why we need a review process :+1: I'll fix it now.

Reply:
Perfect, thanks :)